test/object/object.cc

```diff
@@ -252,6 +252,17 @@ Value CreateObjectUsingMagic(const CallbackInfo& info) {
   return obj;
 }
 
+Value Sum(const CallbackInfo& info) {
+  Array array = info[0].As<Array>();
+  int64_t sum = 0;
+
+  for (const auto& e : array) {
+    sum += e.second.As<Number>().Int64Value();
```

Comment:
Perhaps, I should change this test to iterate over an object because I'm planning on adding a separate iterator for `Napi::Array`, where the `Napi::Array::operator*()` would return the `Value` at the currently iterated index.

Reply:
Having a special iterator for arrays would be nice.